plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -1262,6 +1261,28 @@ public void testDeleteWithVarcharPredicate()
         throw new SkipException("This is implemented by testDeleteWithVarcharEqualityPredicate");
     }
 
+    @Test
+    public void testInsertWithoutTemporaryTable()
+    {
+        if (!hasBehavior(SUPPORTS_CREATE_TABLE)) {
+            throw new SkipException("CREATE TABLE is required for testing non-transactional write support");
+        }
+        Session session = Session.builder(getSession())
+                .setCatalogSessionProperty(getSession().getCatalog().orElseThrow(), "non_transactional_insert", "true")
+                .setCatalogSessionProperty(getSession().getCatalog().orElseThrow(), "insert_batch_size", "16")
```

Comment:
Does this matter? IIUC `beginInsertTable` always uses direct insert now if session property is set.